### PR TITLE
Weapon construction visual upgrade

### PR DESCRIPTION
**Pull request by Fueredoriku**
- Add rotating wrap for item select
- Add icon for no extension
- Add gunstatUI always comparing to the starter weapon instead of previous weapon

### CODE REVIEW

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs [hunk 1]
```diff
@@ -52,55 +60,80 @@ public void SetItems(List<Item> items, Item initialItem)
...
+    }
+
+    private IEnumerator WaitAndSetWeapons()
+    {
+        yield return new WaitForSeconds(0.1f);
+        ChangeItemDisplayed(selectedIndex);
     }
 
-    public void Select()
+    public void Select(out Item item)
```
Comment (toberge) [inline Assets/Scripts/Control&Input/ItemSelectSlot.cs:106]: I'm curious why you chose to use an out parameter instead of changing the return type... And why couldn't you just use the SelectedItem property directly? There's no reason for avoiding so that I can discern

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs [hunk 2]
```diff
@@ -110,15 +143,24 @@ private void ChangeItemDisplayed(int nextIndex, bool animate = true)
...
-        else
+    }
+
+    public static int Mod(float a, float b)
+    {
+        float c = a % b;
+        if ((c < 0 && b > 0) || (c > 0 && b < 0))
         {
-            itemHolder.localPosition = position;
+            c += b;
```
Comment (toberge) [inline Assets/Scripts/Control&Input/ItemSelectSlot.cs:162]: No need to overcomplicate this, simply add `b` to `a` to avoid dropping below 0 in the first place. This method should be removed altogether, or at least turned into an extension method.

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs [hunk 3]
```diff
@@ -52,55 +60,80 @@ public void SetItems(List<Item> items, Item initialItem)
...
     public void Deselect()
     {
         background.color = deselectedColor;
     }
 
-    public void Previous()
+    public void Previous(out Item item)
     {
         // +count to avoid negative numbers from modulo (C# is an oddball here)
         ChangeItemDisplayed((selectedIndex - 1 + items.Count) % items.Count);
```
Comment (toberge) [inline Assets/Scripts/Control&Input/ItemSelectSlot.cs:120]: See this line for how I've solved it in *this very file*

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs [hunk 4]
```diff
@@ -52,55 +60,80 @@ public void SetItems(List<Item> items, Item initialItem)
...
+        LeanTween.moveLocal(tileHolder, itemSlots[5].transform.localPosition, .2f)
```
Comment (toberge) [inline Assets/Scripts/Control&Input/ItemSelectSlot.cs:121]: Um... why is this here and not part of the `for` loop in ChangeItemDisplayed?

File: Assets/Scripts/UI/PlayerStatUI.cs [hunk 5]
```diff
@@ -86,7 +70,7 @@ void OnDestroy()
 
     private void OnInventoryChange(Item item)
     {
-        SetBaseGunStats(GunFactory.GetGunStats(playerIdentity.Body, playerIdentity.Barrel, playerIdentity.Extension));
+        SetBaseGunStats(GunFactory.GetGunStats(StaticInfo.Singleton.StartingBody, StaticInfo.Singleton.StartingBarrel, StaticInfo.Singleton.StartingExtension));
     }
 
     private void OnBiddingPlatformChange(BiddingPlatform platform)
```
Comment (toberge) [inline Assets/Scripts/UI/PlayerStatUI.cs:76]: Hm, we don't really need this one anymore, right?

**Review by toberge — APPROVED** [review]
Ah, now we have something more useful that another view of the gun :tada: 

The ready text and the no-extension-icon perform a sinful ritual that angers the UI gods :triumph:  
image
but I can fix that myself, it's okay

Left some comments but works well otherwise

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs (hunk 1 above)
Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/ItemSelectSlot.cs:106]: It was to avoid convoluted nullable annotation in the return type. My original thought was that there might be cases where one isn't interested in the out variable either, but currently there is no example of that left

Comment (toberge) [reply Assets/Scripts/Control&Input/ItemSelectSlot.cs:106]: But still, why not use the SelectedItem property?

Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/ItemSelectSlot.cs:106]: I'm going to change it to using that property instead, I just forgot to change it since I removed all places where the optional out was relevant 😎

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs (hunk 2 above)
Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/ItemSelectSlot.cs:162]: I can change it to an extension method, an actual modulo operator seems like a useful thing to have instead of only a remainder operator (Thanks c# 😅)

File: Assets/Scripts/UI/PlayerStatUI.cs (hunk 5 above)
Comment (Fueredoriku) [reply Assets/Scripts/UI/PlayerStatUI.cs:76]: Nope, I can remove the depreceated code 👍

File: Assets/Scripts/Control&Input/ItemSelectSlot.cs (hunk 4 above)
Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/ItemSelectSlot.cs:121]: Because I didn't want to animate it on the initial select / not need to the modulo difference for which direction we should move the slots. I also felt like it was slightly easier to read the code like this compared to potentially sending optional bools into the method for deciding direction and wheter to animate or not (which is my prefference, but I concede it might be non-convential in this case)

Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/ItemSelectSlot.cs:113]: I'm gonna leave this as is for now, feel free to refactor later if you still want 😅
